src/app.jsx

```diff
@@ -357,3 +363,124 @@ export const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
         </Select>
     );
 };
+
+const DropdownWithKebab = ({ selected, path }) => {
+    const Dialogs = useDialogs();
+    const [isOpen, setIsOpen] = useState(false);
+
+    const onToggleClick = () => {
+        setIsOpen(!isOpen);
+    };
+    const onSelect = (_event, itemId) => {
+        setIsOpen(false);
+    };
+
+    const deleteItem = () => {
+        const itemPath = "/" + path.join("/") + "/" + selected.name;
+        Dialogs.show(<ConfirmDeletionDialog selected={selected} itemPath={itemPath} />);
+    };
+
+    return (
+        <Dropdown
+            isPlain
+            isOpen={isOpen}
+            onSelect={onSelect}
+            onOpenChange={isOpen => setIsOpen(isOpen)}
+            popperProps={{ position: "right" }}
+            toggle={toggleRef =>
+                <MenuToggle ref={toggleRef} variant="plain" onClick={onToggleClick} isExpanded={isOpen}>
+                    <EllipsisVIcon />
+                </MenuToggle>}
+        >
+            <DropdownList>
+                <DropdownItem itemId='delete-item' key="delete-item" isDisabled={!selected} onClick={deleteItem} className={!selected ? "" : "pf-m-danger"}>
+                    {_("Delete")}
+                </DropdownItem>
+            </DropdownList>
+        </Dropdown>
+    );
+};
+
+const ConfirmDeletionDialog = ({ selected, itemPath }) => {
+    const Dialogs = useDialogs();
+
+    const deleteItem = () => {
+        if (selected.type === "file") {
+            cockpit.spawn(["rm", itemPath]).then(Dialogs.close)
+                    .catch(() => { Dialogs.show(<ForceDeleteModal itemPath={itemPath} />) });
+        } else if (selected.type === "directory") {
+            let emptyDirectory = false;
+            // Check for empty directory
+            cockpit.spawn(["ls", "-A", itemPath])
+                    .then(result => { emptyDirectory = result === "" })
+                    .then(() => {
+                        if (emptyDirectory)
+                            cockpit.spawn(["rmdir", itemPath]).then(Dialogs.close);
+                        else {
+                            cockpit.spawn(["rm", "-r", itemPath]);
+                        }
+                    })
+                    .then(Dialogs.close)
+                    .catch(() => { Dialogs.show(<ForceDeleteModal itemPath={itemPath} />) });
+        }
+    };
+
+    return (
+        <Modal
+            position="top"
+            title={_("Confirm deletion")}
+            isOpen
+            onClose={Dialogs.close}
+            footer={
+                <>
+                    <Button variant='danger' onClick={deleteItem}>{_("Delete Item")}</Button>
+                    <Button variant='link' onClick={Dialogs.close}>{_("Cancel")}</Button>
+                </>
+            }
+        >
+            {cockpit.format(_("Are you sure you want to permenantly delete $0?"), selected.name)}
```

Comment:
Remove the description for now. We will need to do a pass at some point with the designer and it can be introduced then.